pycytominer/cyto_utils/transform.py

```diff
@@ -0,0 +1,54 @@
+"""
+Transform observation variables by specified groups
+"""
+
+import os
+import numpy as np
+import pandas as pd
+from scipy.linalg import eigh
+from sklearn.base import BaseEstimator, TransformerMixin
+
+
+class Whiten(BaseEstimator, TransformerMixin):
+    """
+    Class to whiten data in the base sklearn transform API
+    Note, this implementation is modified from a function written by Juan C. Caicedo
+    """
+
+    def __init__(self, epsilon=1e-18, center=True):
+        """
+        Arguments:
+        epsilon - fudge factor parameter
+        center - option to center input X matrix
+        """
+        self.epsilon = epsilon
+        self.center = center
+
+    def fit(self, X, y=None):
+        """
+        Identify the whitening transform given self.X
+
+        Argument:
+        X - dataframe to fit whitening transform
+        """
+        # Get the mean of the features (columns) and center if specified
+        self.mu = X.mean()
```

Comment:
@gwaygenomics did you mean to write `self.mu = np.mean(X, 0)` ?

Reply:
is it the same thing?

Reply:
`self.mu = X.mean(0)` returns a vector: the mean of `X` along `axis=0`
`self.mu = X.mean()` returns a scalar: the mean of flattened `X`.

Reply:
PS – please don't ruin your weekend pondering this. I added this note so I don't forget. Too bad GitHub doesn't have scheduled send :D

Reply:
> Is X in your example a pandas.DataFrame?

Ah no – it's a numpy array. You're good.